docs/secrets.mdx

```diff
@@ -0,0 +1,630 @@
+```
+
+</CodeExample>
+<CodeExample lang="python">
+
+```
+Secrets support is under development...
+```
+
+</CodeExample>
+<CodeExample lang="java">
+
+```java
+import io.nitric.api.secret.Secrets;
+
+var version = Secrets.secret("database.password").latest();
+```
+
+</CodeExample>
+<CodeExample lang="go">
+
+```go
+import (
+    "github.com/nitrictech/go-sdk/api/secrets"
+)
+
+ss, err := secrets.New()
+
+if err != nil {
+	// handle error
+}
+
+version := ss.Secret("database.password").Latest()
+```
+
+</CodeExample>
+<CodeExample lang="php">
+
+```
+Secrets support is under development...
+```
+
+</CodeExample>
+
+<CodeExample lang="c-sharp">
+
+```csharp
+using Nitric.Api.Secret;
+
+var secrets = new Secrets();
+
+var latest = secrets.Secret("database.password").Latest();
+```
+
+</CodeExample>
+
+</CodeExamples>
+
+Note the `latest` SecretVersion reference is an alias to access the latest SecretValue. Making repeated `access()` calls from
+this latest version may return different secret values if a new version has been created.
+
+To get the actual or materialized SecretVersion for a SecretValue, use SecretValue object's SecretVersion property.
+
+### Access
+
+To access the secret data value of a secret version use the `access()` method.
+
+<CodeExamples 
+	languages={[
+	{
+		label: "Node",
+		value: "node"
+	},
+	{
+		label: "Python",
+		value: "python"
+	},
+	{
+		label: "Java",
+		value: "java"
+	},
+	{
+		label: "Go",
+		value: "go"
+	},
+	{
+		label: "PHP",
+		value: "php"
+	},
+	{
+		label: "C#",
+		value: "c-sharp"
+	}
+	]}
+	defaultLang="node"
+>
+<CodeExample lang="node">
+
+```typescript
+import { storage } from '@nitric/sdk';
+
+const secrets = new Secrets();
+
+const value = secrets.secret('database.password').latest().access();
+
+const password = value.asString();
+```
+
+</CodeExample>
+<CodeExample lang="python">
+
+```
+Secrets support is under development...
+```
+
+</CodeExample>
+<CodeExample lang="java">
+
+```java
+import io.nitric.api.secret.Secrets;
+
+var value = Secrets.secret("database.password")
+	.latest()
+	.access();
+
+var password = value.getAsText();
+```
+
+</CodeExample>
+<CodeExample lang="go">
+
+```go
+import (
+    "github.com/nitrictech/go-sdk/api/secrets"
+)
+
+ss, err := secrets.New()
+
+if err != nil {
+	// handle error
+}
+
+value, err := ss.Secret("database.password")
+	.Latest()
+	.Access()
+
+if err != nil {
+	// handle error
+}
+
+password := value.AsString()
+```
+
+</CodeExample>
+<CodeExample lang="php">
+
+```
+Secrets support is under development...
+```
+
+</CodeExample>
+
+<CodeExample lang="c-sharp">
+
+```csharp
+using Nitric.Api.Secret;
+
+var secrets = new Secrets();
+
+var value = secrets.Secret("database.password")
+	.Latest()
+	.Access();
+
+var password = value.ValueText
+```
+
+</CodeExample>
+
+</CodeExamples>
+
+The SecretValue is a value object containing the secret data. The SecretValue object
+also contains a reference to the actual SecretVersion.
+
+## Examples
```

Comment:
Cool example, might be better as a tutorial that produces a working example.

Reply:
Yeah that would be better, but still useful at this stage to express the concept.  I will raise an nitric/snippy issue.

Reply:
https://github.com/nitrictech/snippy/issues/3